### PR TITLE
Fix Valgrind runs

### PR DESCRIPTION
**Pull request by igchor**

### CODE REVIEW

File: src/memory_targets/memory_target_numa.c [hunk 1]
```diff
@@ -77,12 +77,19 @@ numa_targets_create_nodemask(struct numa_memory_target_t **targets,
         }
     }
 
-    int nrUlongs = hwloc_bitmap_nr_ulongs(bitmap);
-    if (nrUlongs == -1) {
-        hwloc_bitmap_free(bitmap);
-        return UMF_RESULT_ERROR_UNKNOWN;
+    int lastBit = hwloc_bitmap_last(bitmap);
+    if (lastBit == -1) {
+        // no node is set
```
Comment (bratpiorka) [inline src/memory_targets/memory_target_numa.c:82]: you missed hwloc_bitmap_free()

File: src/memory_targets/memory_target_numa.c [hunk 2]
```diff
@@ -77,12 +77,19 @@ numa_targets_create_nodemask(struct numa_memory_target_t **targets,
...
 
+    *maxnode = lastBit + 1;
+
+    // Do not use hwloc_bitmap_nr_ulongs due to:
+    // https://github.com/open-mpi/hwloc/issues/429
+    unsigned bits_per_long = sizeof(unsigned long) * 8;
+    int nrUlongs = (lastBit + bits_per_long) / bits_per_long;
+
     unsigned long *nodemask = malloc(sizeof(unsigned long) * nrUlongs);
     int ret = hwloc_bitmap_to_ulongs(bitmap, nrUlongs, nodemask);
```
Comment (bratpiorka) [inline src/memory_targets/memory_target_numa.c:94]: add a check when nodemask == NULL + handle errors

**Review by bratpiorka — APPROVED** [review]

**Review by ldorau — APPROVED** [review]

**Review by bratpiorka — CHANGES_REQUESTED** [review]
.

**Review by bratpiorka — COMMENTED** [review]
others - please review as I'm on vacation

File: src/memory_targets/memory_target_numa.c (hunk 2 above)
Comment (igchor) [reply src/memory_targets/memory_target_numa.c:85]: Done.

File: src/memory_targets/memory_target_numa.c (hunk 1 above)
Comment (igchor) [reply src/memory_targets/memory_target_numa.c:67]: Done.

Comment (ldorau) [reply src/memory_targets/memory_target_numa.c:67]: hwloc_bitmap_free() added

File: src/memory_targets/memory_target_numa.c (hunk 2 above)
Comment (ldorau) [reply src/memory_targets/memory_target_numa.c:85]: error handling added

**Review by ldorau — CHANGES_REQUESTED** [review]
```
>>>>>>> LOG ./test/umf_test-provider_os_memory_multiple_NUMA_nodes.log
==10975== Memcheck, a memory error detector
==10975== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==10975== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==10975== Command: ./test/umf_test-provider_os_memory_multiple_NUMA_nodes
==10975==
Running main() from /home/ldorau/work/unified-memory-framework/build/_deps/googletest-src/googletest/src/gtest_main.cc
[==========] Running 5 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 5 tests from testNUMAnodesAllocations/testNUMAnodes
[ RUN      ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/0
[       OK ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/0 (13 ms)
[ RUN      ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/1
[       OK ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/1 (0 ms)
[ RUN      ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/2
alloc(1024) = 0x4037000, assigned NUMA node = 1
/home/ldorau/work/unified-memory-framework/test/provider_os_memory_multiple_NUMA_nodes.cpp:94: Failure
Expected equality of these values:
  numa_node
    Which is: 1
  i
    Which is: 0
[  FAILED  ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/2, where GetParam() = (0, 1) (44 ms)
[ RUN      ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/3
binding memory to NUMA node failed: Invalid argument
/home/ldorau/work/unified-memory-framework/test/provider_os_memory_multiple_NUMA_nodes.cpp:89: Failure
Expected equality of these values:
  umf_result
    Which is: 2
  UMF_RESULT_SUCCESS
    Which is: 0
[  FAILED  ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/3, where GetParam() = (1, 1) (14 ms)
[ RUN      ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/4
[       OK ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/4 (0 ms)
[----------] 5 tests from testNUMAnodesAllocations/testNUMAnodes (81 ms total)

[----------] Global test environment tear-down
[==========] 5 tests from 1 test suite ran. (113 ms total)
[  PASSED  ] 3 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/2, where GetParam() = (0, 1)
[  FAILED  ] testNUMAnodesAllocations/testNUMAnodes.checkNUMAnodesAllocations/3, where GetParam() = (1, 1)

 2 FAILED TESTS
==10975==
==10975== HEAP SUMMARY:
==10975==     in use at exit: 1,040 bytes in 16 blocks
==10975==   total heap usage: 485 allocs, 469 frees, 229,690 bytes allocated
==10975==
==10975== 160 (112 direct, 48 indirect) bytes in 1 blocks are definitely lost in loss record 6 of 8
==10975==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==10975==    by 0x121AC2: umfMemoryProviderCreate (memory_provider.c:30)
==10975==    by 0x11458E: testNUMAnodes_checkNUMAnodesAllocations_Test::TestBody() (provider_os_memory_multiple_NUMA_nodes.cpp:80)
==10975==    by 0x163D78: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (gtest.cc:2599)
==10975==    by 0x15C452: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (gtest.cc:2635)
==10975==    by 0x1377E1: testing::Test::Run() (gtest.cc:2674)
==10975==    by 0x1382B4: testing::TestInfo::Run() (gtest.cc:2853)
==10975==    by 0x138BF2: testing::TestSuite::Run() (gtest.cc:3012)
==10975==    by 0x148E24: testing::internal::UnitTestImpl::RunAllTests() (gtest.cc:5870)
==10975==    by 0x164DFF: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (gtest.cc:2599)
==10975==    by 0x15D66A: bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (gtest.cc:2635)
==10975==    by 0x147446: testing::UnitTest::Run() (gtest.cc:5444)
==10975==
==10975== 720 (80 direct, 640 indirect) bytes in 5 blocks are definitely lost in loss record 8 of 8
==10975==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==10975==    by 0x486956A: numa_bitmask_alloc (in /usr/lib/x86_64-linux-gnu/libnuma.so.1.0.0)
==10975==    by 0x11445F: testNUMAnodes_checkNUMAnodesAllocations_Test::TestBody() (provider_os_memory_multiple_NUMA_nodes.cpp:74)
==10975==    by 0x163D78: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (gtest.cc:2599)
==10975==    by 0x15C452: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (gtest.cc:2635)
==10975==    by 0x1377E1: testing::Test::Run() (gtest.cc:2674)
==10975==    by 0x1382B4: testing::TestInfo::Run() (gtest.cc:2853)
==10975==    by 0x138BF2: testing::TestSuite::Run() (gtest.cc:3012)
==10975==    by 0x148E24: testing::internal::UnitTestImpl::RunAllTests() (gtest.cc:5870)
==10975==    by 0x164DFF: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (gtest.cc:2599)
==10975==    by 0x15D66A: bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (gtest.cc:2635)
==10975==    by 0x147446: testing::UnitTest::Run() (gtest.cc:5444)
==10975==
==10975== LEAK SUMMARY:
==10975==    definitely lost: 192 bytes in 6 blocks
==10975==    indirectly lost: 688 bytes in 7 blocks
==10975==      possibly lost: 0 bytes in 0 blocks
==10975==    still reachable: 160 bytes in 3 blocks
==10975==         suppressed: 0 bytes in 0 blocks
==10975== Reachable blocks (those to which a pointer was found) are not shown.
==10975== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==10975==
==10975== For lists of detected and suppressed errors, rerun with: -s
==10975== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: 0 from 0)
```

**Review by PatKamin — APPROVED** [review]

**Review by ldorau — APPROVED** [review]

**Review by kswiecicki — APPROVED** [review]

**Pull request merged**